### PR TITLE
Updated README

### PR DESCRIPTION
**Pull request by boczkowska**
Signed-off-by: Yana Hontyk <yhontyk@redhat.com>

**What type of PR is this?**

/kind documentation
[skip ci]

**What does does this PR do / why we need it**:
Updates README to ensure OpenShift and Kubernetes are mentioned together. Fixes a few small bugs (unresolved attributes, probably inherited from the downstream docs at some point.)
**Which issue(s) this PR fixes**:

Fixes #?

**How to test changes / Special notes to the reviewer**:

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -80,13 +80,13 @@ The list of available container images is sourced from the cluster's internal co
 
 To list the available components and associated container images for your cluster:
 
-. Log in to the {product-title} cluster with {odo-title}:
+. Log in to the cluster with `odo`:
```
Comment (dharmit) [inline README.adoc:83]: This is OpenShift specific. One can't login to the k8s cluster.

Comment (kadel) [reply README.adoc:83]: Yes, and no :-) 
Even Kubernetes cluster can have authentication but we currently don't support that for Kubernetes.

Comment (dharmit) [reply README.adoc:83]: Yeah. My comment was based on `odo login` always failing for me when I am working with minikube cluster. :smile:

Comment (boczkowska) [reply README.adoc:83]: WDYT about: "Access your cluster. For OpenShift cluster, you can authenticate with `odo login`. 
Authentication with `odo login` for Kubernetes clusters is currently not supported."

Comment (dharmit) [reply README.adoc:83]: @boczkowska Makes sense to me. In my experience, `odo` commands kind of just work with kubernetes. Login is not required for it.

Comment (boczkowska) [reply README.adoc]: @dharmit reworded.

File: README.adoc [hunk 2]
```diff
@@ -80,13 +80,13 @@ The list of available container images is sourced from the cluster's internal co
 
 To list the available components and associated container images for your cluster:
 
-. Log in to the {product-title} cluster with {odo-title}:
+. Access your cluster. Authentication with odo login for Kubernetes clusters is currently not supported. For OpenShift cluster, you can authenticate with `odo login`:
```
Comment (dharmit) [inline README.adoc:83]: Please change it to "`odo login` for Kubernetes..."

Comment (dharmit) [reply README.adoc:83]: @boczkowska sorry about the nitpick. But I think `odo login` makes more sense. :)

Comment (boczkowska) [reply README.adoc]: @dharmit done. :relaxed:

**Pull request merged**